### PR TITLE
Add Perfecto cloud to the Appium desktop recorder

### PR DESCRIPTION
**Pull request by UziEilon**
Added Perfecto to desktop Appium UI.
This will allow users to record directly from Perfecto cloud

### CODE REVIEW

File: app/renderer/actions/Session.js [hunk 1]
```diff
@@ -208,9 +210,25 @@ export function newSession (caps, attachSessId = null) {
       case ServerTypes.headspin:
         host = session.server.headspin.hostname;
         port = session.server.headspin.port;
-        path = `/v0/${session.server.headspin.apiKey}/wd/hub`;
+        path = `/wd/hub`;
```
Comment (imurchie) [inline app/renderer/actions/Session.js]: Why are you changing things for Headspin's integration?

File: app/renderer/actions/Session.js [hunk 2]
```diff
@@ -208,9 +210,25 @@ export function newSession (caps, attachSessId = null) {
...
         break;
+      case ServerTypes.perfecto:
+        host = session.server.perfecto.hostname;
+        port = session.server.perfecto.port;
+        token = session.server.perfecto.token;
+        path = "/nexperience/perfectomobile/wd/hub";
+        if (!token) {
+          notification.error({
+            message: "Error",
+            description: "Perfecto Token is require!",
```
Comment (imurchie) [inline app/renderer/actions/Session.js]: Grammar.

File: app/renderer/actions/Session.js (hunk 1 above)
Comment (UziEilon) [reply app/renderer/actions/Session.js]: my mistake, will fix it now

**Comment by UziEilon** [comment]
revert the Headspin change

File: .idea/appium-desktop.iml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module type="JAVA_MODULE" version="4">
+  <component name="NewModuleRootManager" inherit-compiler-output="true">
+    <exclude-output />
+    <content url="file://$MODULE_DIR$" />
+    <orderEntry type="inheritedJdk" />
+    <orderEntry type="sourceFolder" forTests="false" />
+  </component>
+</module>
```
Comment (dpgraham) [inline .idea/appium-desktop.iml]: Do we need these `.idea` files?

File: app/renderer/components/Session/Session.js [hunk 4]
```diff
@@ -12,6 +12,7 @@ import ServerTabHeadspin from './ServerTabHeadspin';
 import ServerTabBrowserstack from './ServerTabBrowserstack';
 import ServerTabBitbar from './ServerTabBitbar';
 import ServerTabKobiton from './ServerTabKobiton';
+import SereverTabPerfecto from './ServerTabPerfecto';
```
Comment (KazuCocoa) [inline app/renderer/components/Session/Session.js]: Typo? s/serever/server/

**Comment by UziEilon** [comment]
image

Perfecto tab

**Comment by imurchie** [comment]
The layout of these tabs is getting too cumbersome. There should be no scrolling of it.

File: app/renderer/actions/Session.js [hunk 5]
```diff
@@ -211,6 +213,22 @@ export function newSession (caps, attachSessId = null) {
         path = `/v0/${session.server.headspin.apiKey}/wd/hub`;
         https = true;
         break;
+      case ServerTypes.perfecto:
+        host = session.server.perfecto.hostname;
+        port = session.server.perfecto.port;
+        token = session.server.perfecto.token;
```
[the remark this answers is not in the document]
Comment (UziEilon) [reply app/renderer/actions/Session.js]: i added support in env parameter so I kept it let

**Comment by UziEilon** [comment]
imurchie - Can you please explain the scrolling comments?

Did you talk about the README table scrolling ? 
I checked it and it looks good, see below: 
image

or on the main screen tabs , and in this case you have the scrolling today 
image

File: .idea/appium-desktop.iml (hunk 3 above)
Comment (UziEilon) [reply .idea/appium-desktop.iml]: done

File: app/renderer/actions/Session.js [hunk 6]
```diff
@@ -295,6 +313,9 @@ export function newSession (caps, attachSessId = null) {
       showError(e, 0);
     });
 
+
+
+
```
[the remark this answers is not in the document]
Comment (UziEilon) [reply app/renderer/actions/Session.js]: Done

**Comment by dpgraham** [comment]
@UziEilon He just means that the tabs are getting too cluttered with cloud providers and that we need to reorganize how they're disabled We can still accept this PR as it is.

**Comment by imurchie** [comment]
Dan is correct. Though I'd prefer to defer this until the layout is resolved.

File: .idea/appium-desktop.iml (hunk 3 above)
Comment (KazuCocoa) [reply .idea/appium-desktop.iml]: You should also git delete `.idea/` files you have committed to

**Comment by dpgraham** [comment]
@UziEilon one last thing is to delete the `.idea` files... once that's done I can merge this PR.

**Pull request merged**